### PR TITLE
[Task 1865842] initialize storage lib module and add storage related models.

### PR DESCRIPTION
**Pull request by shenqianjin**
[AB#1865842](https://dev.azure.com/mseng/a4d27ce2-a42d-4b71-8eef-78cee9a9728e/_workitems/edit/1865842)

### CODE REVIEW

File: azure-toolkit-libs/azure-toolkit-storage-lib/src/main/java/com/microsoft/azure/toolkit/lib/storage/model/KindEnum.java [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+ */
+
+package com.microsoft.azure.toolkit.lib.storage.model;
+
+import lombok.AllArgsConstructor;
+import lombok.Getter;
+
+@Getter
+@AllArgsConstructor
+public enum KindEnum {
```
Comment (andxu) [inline azure-toolkit-libs/azure-toolkit-storage-lib/src/main/java/com/microsoft/azure/toolkit/lib/storage/model/KindEnum.java:13]: Kind

File: azure-toolkit-libs/azure-toolkit-storage-lib/src/main/java/com/microsoft/azure/toolkit/lib/storage/model/KindEnum.java [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+
+    Storage("Storage", "General Purpose v1"),
+    StorageV2("StorageV2", "General Purpose v2"),
+    BlobStorage("BlobStorage", "Blob Storage"),
+    FileStorage("FileStorage", "File Storage"),
+    BlockBlobStorage("BlockBlobStorage", "Block Blobs Storage");
+
+    private String code;
```
Comment (andxu) [inline azure-toolkit-libs/azure-toolkit-storage-lib/src/main/java/com/microsoft/azure/toolkit/lib/storage/model/KindEnum.java:21]: name + label

File: azure-toolkit-libs/azure-toolkit-storage-lib/src/main/java/com/microsoft/azure/toolkit/lib/storage/model/StorageAccountEntity.java [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+
+    public Region getRegion() {
+        return remoteOptional().map(remote -> Region.fromName(remote.regionName())).orElse(null);
+    }
+
+    @Override
+    public String getPerformance() {
+        return remoteOptional().map(remote -> {
+            String[] replicationArr = remote.skuType().name().toString().split("_");
+            return replicationArr.length == 2 ? replicationArr[0] : null;
```
Comment (andxu) [inline azure-toolkit-libs/azure-toolkit-storage-lib/src/main/java/com/microsoft/azure/toolkit/lib/storage/model/StorageAccountEntity.java:52]: NPE?

File: azure-toolkit-libs/azure-toolkit-storage-lib/src/main/java/com/microsoft/azure/toolkit/lib/storage/model/StorageAccountConfig.java [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+@Builder
+public class StorageAccountConfig implements IStorageAccountEntity {
+
+    private String name;
+    private String id;
+    private ResourceGroup resourceGroup;
+    private Subscription subscription;
+    private Region region;
+
+    private String performance;
```
Comment (andxu) [inline azure-toolkit-libs/azure-toolkit-storage-lib/src/main/java/com/microsoft/azure/toolkit/lib/storage/model/StorageAccountConfig.java:28]: enum

File: azure-toolkit-libs/azure-toolkit-storage-lib/src/main/java/com/microsoft/azure/toolkit/lib/storage/model/KindEnum.java (hunk 1 above)
Comment (shenqianjin) [reply azure-toolkit-libs/azure-toolkit-storage-lib/src/main/java/com/microsoft/azure/toolkit/lib/storage/model/KindEnum.java]: renamed.

File: azure-toolkit-libs/azure-toolkit-storage-lib/src/main/java/com/microsoft/azure/toolkit/lib/storage/model/KindEnum.java (hunk 2 above)
Comment (shenqianjin) [reply azure-toolkit-libs/azure-toolkit-storage-lib/src/main/java/com/microsoft/azure/toolkit/lib/storage/model/KindEnum.java]: updated.

File: azure-toolkit-libs/azure-toolkit-storage-lib/src/main/java/com/microsoft/azure/toolkit/lib/storage/model/StorageAccountConfig.java (hunk 4 above)
Comment (shenqianjin) [reply azure-toolkit-libs/azure-toolkit-storage-lib/src/main/java/com/microsoft/azure/toolkit/lib/storage/model/StorageAccountConfig.java]: updated.

**Review by andxu — APPROVED** [review]

**Comment by andxu** [comment]
LGTM

**Pull request merged**